Some assembly requaired

Razharas opened the pull request:
Changes wirecutters for an igniter in a stunprod
Stunprod sparkles on proding
Timer now have "repeating" setting, will restart again and again after
expiring
Voice analyzer got exclusive mode that works like a password, i.,e. need
a perfect match of the phrase
Voice analyzer got voice recognition mode that remembers human voice
instead of speech, works like personality recognition
Voice analyzer got a sensor mode, works on any speech it hears from
anyone
Voice analyzer modes and changed by applying a multitool to it

Razharas committed:
Some assembly requaired

Changes wirecutters for an igniter in a stunprod
Stunprod sparkles on proding
Timer now have repeating setting, will restart again and again after
expiring
Voice analyzer got exclusive mode that works like a password, i.,e. need
a perfect match of the phrase
Voice analyzer got voice recognition mode that remembers human voice
instead of speech, works like personality recognition
Voice analyzer got a sensor mode, works on any speech it hears from
anyone
Voice analyzer modes and changed by applying a multitool to it

Iamgoofball commented:
can the voice analyzer pick up voice changer fakes as the real deal with voice recognition mode?

Lumipharon commented:
The stun prod change makes literally no sense.

Razharas commented:
Nope, it just saved mob and checks it
Suggest alternatives

Buggy123 commented:
Nice but stunprod change is unnecessary and bizarre. In what way does a stun prod use a IGNITER.

Iamgoofball commented:
have the voice analyzer check the actual name supplied, I'm pretty sure Hear() has some shit where it passes that

paging @miauw62 the saycode expert

DaedalusGame commented:
le igniter producing sparks meme

MMMiracles commented:
>requaired

Buggy123 commented:
Yeah I guess a igniter makes like a burst of high voltage or something and thats why it sparks etc but its dumb, wirecutters make perfect sense.

Razharas commented:
Im just trying to push some usability upon assemblies guise dont appeal to realism plz

Razharas commented:
The name match sounds interesting

PKPenguin321 commented:
:-1: to the stunprod change, it's dumb and makes no sense and seems to be change for the sake of change
other stuff is cool though

KazeEspada commented:
Why not both igniters and wirecutters!

PKPenguin321 commented:
>So this changes nothing.

not necessarily, i often have a pair of wirecutters in my toolbelt for making prods on the fly if the need arises, an igniter not so much
besides, it's really just a change for the sake of change. there's really no point in doing it

Lumipharon commented:
why does an igniter have anything to do with how a prod works?

Razharas committed:
New proximity code

New proximity code

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Assembly

duncathan commented:
Igniters use sparks to ignite so naturally they would have two conducive ends through which a current can flow; just like wirecutters do. 
The point of the change is to make usage of assemblies more prevalent. 

Seems pretty obvious to me.

PKPenguin321 commented:
i dont get why its necessary to change to igniters though
like why

duncathan commented:
>The point of the change is to make usage of assemblies more prevalent.

Lumipharon commented:
how does it make the use of assemblies more prevalent?

An igniter is not an assembly. It's an igniter, which logically would have far smaller components with which you could actually zap someone with, as opose to a big ass set of wire cutters.

MMMiracles commented:
It just occurred to me that you never bother to strip off the insulation on the grip of the wirecutters when building a stunprod.

please fix.

Lumipharon commented:
they're not insulated though. Try snip snip some wires without gloves.

Just like wires, NT probably just painted them to look insulated.

DaedalusGame commented:
they're insulated in highly conductive polymerized plasma.

neersighted commented on code/modules/assembly/voice.dm:
```diff
@@ -11,20 +11,65 @@
 	var/listening = 0
 	var/recorded = "" //the activation message
+	var/mode = 1
+	var/global/list/modes = list("inclusive",
```
defines pls

Miauw62 replied:
this ^^^^^ ^^^ ^6 ^^

Miauw62 commented:
WHO SUMMONS ME

Miauw62 commented on code/modules/assembly/voice.dm:
```diff
@@ -11,20 +11,65 @@
 
+/obj/item/device/assembly/voice/proc/record_speech(atom/movable/speaker, raw_message)
+	switch(mode)
+		if(1)
```
ew

Miauw62 commented on code/modules/assembly/voice.dm:
```diff
@@ -11,20 +11,65 @@
+			if(raw_message == recorded)
+				. = 1
+		if(3)
+			if(speaker == recorded)
```
this doesnt work with radios, pls fix

neersighted commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -117,7 +117,7 @@
 		qdel(src)
 
-	else if(istype(I, /obj/item/weapon/wirecutters) && !(I.flags & NODROP))
+	else if(istype(I, /obj/item/device/assembly/igniter) && !(I.flags & NODROP))
```
Remove nodrop check.

neersighted commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -125,7 +125,7 @@
 		user.unEquip(I)
```
```
if(!user.drop_item())
    return
```
Checks nodrop for you

neersighted commented on code/game/objects/items/weapons/weaponry.dm:
```diff
@@ -117,7 +117,7 @@
+	else if(istype(I, /obj/item/device/assembly/igniter) && !(I.flags & NODROP))
 		var/obj/item/weapon/melee/baton/cattleprod/P = new /obj/item/weapon/melee/baton/cattleprod
 
 		if(!remove_item_from_storage(user))
```
We don't use this anymore.

Razharas committed:
Further impovements to proximity code

Further impovements to proximity code
Also voice analiser's third mode now remembers the voice properly not
just mob

Razharas committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into Assembly

Razharas commented:
I can buff batons in some way if you think using igniter is a nerf

Lumipharon commented:
It's not a nerf, it's just a pointless change.

It's like if you changed flamethrowers to use a wrench instead of a welder. It doesn't make any sense and has no balance effect. It's change for the sake of change.

PKPenguin321 commented:
^

Razharas commented:
Its a change to make igniters have some uses outside stuff you can use only as antag

PKPenguin321 commented:
why bother though, we have plenty of things with antag-only uses

PKPenguin321 commented:
I CHANGED MY MIND THIS IS SOME GOOD SHIT
:+1:👀:+1:👀:+1:👀:+1:👀:+1:👀 good shit go౦ԁ sHit:+1: thats ✔ some good:+1::+1:shit right:+1::+1:there:+1::+1::+1: right✔there ✔✔if i do ƽaү so my self 💯 i say so 💯 thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ💯 :+1::+1: :+1:НO0ОଠOOOOOОଠଠOoooᵒᵒᵒᵒᵒᵒᵒᵒᵒ:+1: :+1::+1: :+1: 💯 :+1: 👀 👀 👀 :+1::+1:Good shit

Lumipharon commented:
>Its a change to make igniters have some uses outside stuff you can use only as antag
>flamethrowers can never be used by non antags
>grenades can never be used by non antags
>ore redemption machines are never built by non antags

And all that aside, is having an item that isn't commonly used by non antags even an issue? Like really? Lets make tyrant and antimov boards glow so non antags can use them as flashlights :^)

duncathan commented:
Clearly no amount of obvious reasoning will get to you two

phil235 commented:
no changelog?

Razharas commented:
Dunc merged it so fast i neither added all the things i wanted nor been able to make changelog
I can add it in the second pr

duncathan commented:
My bad
I'm gonna take a break from merging things for a day or two I need to cool off

Lumipharon commented:
Still no one has explained an actual reason for the prod changes other than "I want a completely unrelated item to have more uses".
